Matrix cleanup 4.5

siefkenj opened the pull request:
This builds on #12390 by deprecating the determinant methods talked about in #12389 

Introduced `MatrixDeprecated` to house deprecated matrix
methods until they are ready for removal.  Methods that have
been deprecated are:

		berkowitz
		berkowitz_det
		berkowitz_eigenvals
		berkowitz_minors
		berkowitz_charpoly
		cofactorMatrix
		det_bareiss
		det_LU_decomposition
		minorEntry
		minorMatrix

They have been replaced with the methods

		det
		charpoly
		eigenvals
		cofactor_matrix
		minor
		minor_submatrix

siefkenj committed:
Deprecate Matrix methods

	Introduced `MatrixDeprecated` to house deprecated matrix
	methods until they are ready for removal.  Methods that have
	been deprecated are:

		berkowitz
		berkowitz_det
		berkowitz_eigenvals
		berkowitz_minors
		berkowitz_charpoly
		cofactorMatrix
		det_bareiss
		det_LU_decomposition
		minorEntry
		minorMatrix

	They have been replaced with the methods

		det
		charpoly
		eigenvals
		cofactor_matrix
		minor
		minor_submatrix

siefkenj committed:
Merge branch 'commonmatrix2' into commonmatrix5

smichr commented:
I see the @moorepants merged "Phase 1". What is the next thing to merge and is there a preferred person to look at it and merge it?

siefkenj commented:
Step 4 ( PR #12390 ) is the next thing to merge.  I don't know if there is a preferred person to merge it, but when things get merged, I can start Step 5, which would be separating out the eigen-vector-related operations and the row-reduction-based operations.

siefkenj committed:
Merge remote-tracking branch 'upstream/master' into commonmatrix5

siefkenj committed:
Replace bodies of deprecated functions

siefkenj committed:
Removed docstrings from deprecated methods

	The docstrings for `berkowitz_charpoly` and `berkowitz` were
	causing the `sympy.doctest` to fail.

asmeurer commented:
@smichr no one is preferred to merge it. If you feel these refactor PRs are OK to merge, feel free to do so. I don't want @siefkenj to get blocked for lack of reviewing.

siefkenj commented:
@jksuom @smichr would you mind taking a look at this PR?

jksuom commented on sympy/matrices/matrices.py:
```diff
@@ -2040,7 +2044,154 @@ def minor_submatrix(self, i, j):
+
+    @deprecated(useinstead="det(method='bareiss')", issue=12363, deprecated_since_version="1.1")
+    def det_bareis(self):
+        return self.det(method='bariess')
```
bareiss

jksuom commented:
Looks good to me (with typo fixed).

siefkenj committed:
Fixed typeos

jksuom commented:
This is ready. Thanks!
